onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1743,10 +1737,10 @@ void CreateInferencePybindStateModule(py::module& m) {
     import_array1();
   })();
 
-  Environment& env = GetEnv();
+  auto env = GetEnv();
```

Comment:
So you intentionally want to create a new shared_ptr instead of taking a reference?

Reply:
Yes, that's the interface. Return shared_ptr by value, meaning ref-count it. This way, it is very deterministic. We can detect that the static is already gone and error out, or we get the ref-count and we are fine.